dsl/src/main/scala/com.crobox.clickhouse/dsl/execution/ClickhouseQueryExecutor.scala

```diff
@@ -29,7 +31,8 @@ trait ClickhouseQueryExecutor extends QueryExecutor { self: TokenizerModule =>
 
   def executeWithProgress[V: JsonReader](
       query: Query
-  )(implicit executionContext: ExecutionContext): Source[ClickHouseExecutor.QueryProgress, Future[QueryResult[V]]] =
+  )(implicit executionContext: ExecutionContext,
+    settings: QuerySettings = QuerySettings()): Source[QueryProgress, Future[QueryResult[V]]] =
```

Comment:
Should we provide a default-value for the settings?

Reply:
The default settings are empty and have a fallback to config settings, so it makes sense to provide this default as in most use cases you do not want to override the config settings